File: app/src/main/java/cn/winxo/toolbox/module/contract/HomeContract.java
package cn.winxo.toolbox.module.contract;

import cn.winxo.toolbox.base.ibase.IPresenter;
import cn.winxo.toolbox.base.ibase.IView;
import cn.winxo.toolbox.data.entity.local.Task;
import java.util.List;

public interface HomeContract {

  interface View extends IView {

    void showData(List<Task> tasks);

    void addTask(Task task);

    void removeTask(int position);
  }

  interface Presenter extends IPresenter<View> {

    void loadTask();

    void addTask(String content);

    void removeTask(int position, Task task);
  }
}

File: app/src/main/java/cn/winxo/toolbox/module/contract/HomeContract.kt
package cn.winxo.toolbox.module.contract

import cn.winxo.toolbox.base.ibase.IPresenter
import cn.winxo.toolbox.base.ibase.IView
import cn.winxo.toolbox.data.entity.local.Task

interface HomeContract {

    interface View : IView {

        fun showData(tasks: List<Task>)

        fun addTask(task: Task)

        fun removeTask(position: Int)
    }

    interface Presenter : IPresenter<View> {

        fun loadTask()

        fun addTask(content: String)

        fun removeTask(position: Int, task: Task)
    }
}